src/platform/pressureFlowSensor.c

```diff
@@ -0,0 +1,93 @@
+#include "stm32f4xx_hal.h"
+#include <platform/pressureFlowSensor.h>
+#include <platform/pressureFlowSensor_private.h>
+#include <platform/configuration_private.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+
+uint32_t read_FS6122_sensor(uint8_t i2c_addr, uint8_t reg_addr){
+    HAL_StatusTypeDef ret;
+    uint8_t buf[20];
+    uint32_t val; 
+
+    // Tell FS6122 which register to read from (pressure, flow, temp, RH)
+    buf[0] = reg_addr;
+    ret = HAL_I2C_Master_Transmit(&i2c1_bus, i2c_addr, buf, 1, HAL_MAX_DELAY);
+    if ( ret != HAL_OK ) {
+      strcpy((char*)buf, "Error Tx\r\n");
+      val = 128;
+    } 
+    else {
+      // Read 4 bytes from the pressure register
+      ret = HAL_I2C_Master_Receive(&i2c1_bus, i2c_addr, buf, 4, HAL_MAX_DELAY);
+      if ( ret != HAL_OK ) {
+        strcpy((char*)buf, "Error Rx\r\n");
+        val = 256;
+      } 
+      else {
+
+        //Combine the bytes
+        val = ((int16_t)buf[0] << 24) | (buf[1] << 16) | (buf[2] << 8) | (buf[3]) >> 2; // value is 14 bit, so shift by 2
```

Comment:
`buf` is overloaded here. In one spot it used for as a UART transmission buffer and in other its for tx/rx over i2c. I think this is a bit confusing. We aren't pressed to save memory here, so let's try to make the code easier to understand and use separate buffers.